relay-event-normalization/src/normalize/processor.rs

```diff
@@ -878,19 +878,22 @@ fn normalize_performance_score(
                 if !profile
                     .score_components
                     .iter()
-                    .all(|c| measurements.contains_key(c.measurement.as_str()))
+                    .all(|c| measurements.contains_key(c.measurement.as_str()) || c.weight == 0.0)
```

Comment:
So this float comparison is tricky. While the comparison here actually does work in most cases (even for `-0.0 == 0.0`) we do not actually know the value of `c.weight`. If weight is purely deserialized this code will probably work as intended, but the moment it is a result of a calculation, we cannot be sure 0 is actually 0.

Example:

```rs
let f = 0.3f32 / (0.9f32 / 3f32) - 1.0f32;
println!("{:.32} is 0? {}", f, f == 0.0f32);
```

This is trivially correct, but the code prints false.

Reply:
Agree with this, though we do `==` comparisons with floats parsed from config in other places as well, e.g.

https://github.com/getsentry/relay/blob/b239d2b09c9fcf39ccc40c4d0ac9b9cd8bd812ef/relay-sampling/src/evaluation.rs#L280

Reply:
Solution for 0 is actually pretty short: `f.abs() <= f32::EPSILON`

Reply:
Updated to make use of `f64::EPSILON`